python/pyspark/conf.py

```diff
@@ -101,13 +101,25 @@ def __init__(self, loadDefaults=True, _jvm=None, _jconf=None):
             self._jconf = _jconf
         else:
             from pyspark.context import SparkContext
-            SparkContext._ensure_initialized()
             _jvm = _jvm or SparkContext._jvm
-            self._jconf = _jvm.SparkConf(loadDefaults)
+
+            if _jvm:
+                # JVM is created, so create self._jconf directly through JVM
+                self._jconf = _jvm.SparkConf(loadDefaults)
+                self._conf = None
+            else:
+                # JVM is not created, so store data in self._conf first
+                self._jconf = None
+                self._conf = {}
 
     def set(self, key, value):
         """Set a configuration property."""
-        self._jconf.set(key, unicode(value))
+        # Try to set self._jconf first if JVM is created, set self._conf if JVM is not created yet.
+        if self._jconf:
```

Comment:
minor:  I think it's generally better and would be more consistent with the rest of the code if you made these checks against a `None` value, here and other places in this PR.  For example:
```python
if self._jconf is not None:
```